cantools/db/formats/dbc.py

```diff
@@ -374,17 +374,19 @@ def _dump_messages(database):
     for message in database.messages:
         msg = []
         fmt = 'BO_ {frame_id} {name}: {length} {senders}'
-        msg.append(fmt.format(frame_id=message.frame_id,
+        msg.append(fmt.format(frame_id=message.frame_id | (1 << 31 if message.is_extended_frame else 0),
```

Comment:
This line is over  80-90 characters long. Either add line breaks, or create a local function called `get_frame_id(message)` and put the logic in that.